datalad_container/containers_add.py

```diff
@@ -170,6 +204,14 @@ def __call__(name, url=None, dataset=None, call_fmt=None, image=None):
         to_save = []
         imgurl = url
         if url:
+            if update and op.lexists(image):
+                # XXX: check=False is used to avoid dropping the image. It
+                # should use drop=False if remove() gets such an option (see
+                # DataLad's gh-2673).
+                for r in ds.remove(image, save=False, check=False,
```

Comment:
Hm, iirc remove also drops that file content, so we might loose that previous version of image forever. I would have just used regular gitrepo remove for now here

Reply:
> Hm, iirc remove also drops that file content

I don't think that is true. 

```
% datalad run "echo blah >blah"
% datalad remove --nocheck blah
remove(ok): blah
save(ok): /tmp/tmp.Vn2kcJWMVw-dl (dataset)
action summary:
  remove (ok: 1)
  save (ok: 1)
% git-annex findref master~
blah
```

Reply:
hm, indeed... https://github.com/datalad/datalad/issues/2673#issuecomment-400686307  - so not dropped if `--nocheck`.  really obscure but in effect correct here, so could be accepted as is I guess for now. Thanks

Reply:
To be fair, this change came with a comment that points datalad/datalad#2673 and mentions the reason for the weirdness (it's even visible from your inline comment).